test/jpypetest/test_array.py

```diff
@@ -601,3 +602,24 @@ def testShortcut(self):
     def testJArrayIndex(self):
         with self.assertRaises(TypeError):
             jpype.JArray[10]
+
+    def testJArrayGeneric(self):
+        self.assertEqual(type(JObject[1]), JArray[JObject])
+
+    def testJArrayGeneric_Init(self):
+        Arrays = JClass("java.util.Arrays")
+        self.assertTrue(Arrays.equals(JObject[0], JArray[JObject](0)))
+
+    def testJArrayInvalidGeneric(self):
+        with self.assertRaises(TypeError):
+            jpype.JArray[object]
+    
+    def testJArrayGenericJClass(self):
+        self.assertEqual(type(JClass[0]), JArray[JClass])
+
+    def testJArrayGenericString(self):
+        self.assertEqual(type(JClass[0]), JArray["java.lang.Class"])
```

Comment:
This one is wild to me - it is like a back-door to ``jp.JClass(...)``.

Reply:
I just used it as a test for a "forward declaration". It could have been any valid class.

Reply:
It has to be any valid class available from the namespace in which it is declared. While `java` in this case is valid from this module, it isn't in general.

A better test would be to create a never-before-seen alias, and use that in the test ``from jpype import java as _java_alias`` and use it in `JArray["_java_alias.lang.Class"]`. If following the PEP, it would be expected that this works, but I don't think it will with your current implementation. 

Honestly, I think it is better to rely on ``from __future__ import annotations`` to avoid having to introduce a new API for `JClass` retrieval (i.e. don't support strings directly, since they aren't a short-cut for avoiding imports).

Reply:
I had some thoughts about using `typing.ForwardRef` because it could be used before the JVM is started. Then it occurred to me that you are correct and just not supporting strings directly is the correct approach.

The main reason I am convinced that not directly supporting strings is correct is because, if you wanted to do it before the JVM is started, all you have to do is the following:

```py
from jpype import JArray

def example(obj: "JArray[java.lang.Object]"):
    ...

```